### PR TITLE
App Rule Enpoints

### PR DESCRIPTION
**Pull request by memosys**
Not intending to merge yet, looking for feedback on endpoints/verbs

### CODE REVIEW

File: web-portal/backend/src/apps/apps.controller.ts [hunk 1]
```diff
@@ -27,4 +37,44 @@ export class AppsController {
     const { name, description } = createAppDto;
     return this.appsService.createApp(userAddress, name, description);
   }
+
+  @Delete(':appId')
+  async deleteApp(@Param('appId') appId: string) {
+    // @note: This action deletes app for given appId;
+    return this.appsService.deleteApp(appId);
+  }
+
```
[the remark this answers is not in the document]
Comment (memosys) [reply web-portal/backend/src/apps/apps.controller.ts:46]: Are you sure? this is to delete an entity

[the remark this answers is not in the document]
Comment (plor) [reply web-portal/backend/src/apps/apps.controller.ts:46]: I'll redo to avoid confusion.

File: web-portal/backend/src/apps/apps.controller.ts [hunk 2]
```diff
@@ -27,4 +37,44 @@ export class AppsController {
...
+  @Patch(':appId')
```
Comment (plor) [inline web-portal/backend/src/apps/apps.controller.ts:47]: probably should be PUT instead.

File: web-portal/backend/src/apps/apps.controller.ts [hunk 3]
```diff
@@ -27,4 +37,44 @@ export class AppsController {
...
+  async updateApp(@Param('appId') appId: string, @Body() updateAppDto: any) {
+    // @note: This action updates app for given appId;
+    return this.appsService.updateApp(appId, updateAppDto);
+  }
+
+  @Put(':appId/rule')
```
Comment (plor) [inline web-portal/backend/src/apps/apps.controller.ts:53]: Should be POST

File: web-portal/backend/src/apps/apps.controller.ts [hunk 4]
```diff
@@ -27,4 +37,44 @@ export class AppsController {
...
+  async createAppRule(
+    @Param('appId') appId: string,
+    @Body() createAppRuleDto: any,
+  ) {
+    // @note: This action creates app rule given appId;
+    return this.appsService.createAppRule(appId, createAppRuleDto);
+  }
+
+  @Patch(':appId/rule/:ruleId')
```
Comment (plor) [inline web-portal/backend/src/apps/apps.controller.ts:62]: Probably PUT instead.

**Review by plor — COMMENTED** [review]
Sorry for the mix-up

**Pull request merged**